crates/runc-shim/src/runc.rs

```diff
@@ -349,6 +349,56 @@ impl ProcessLifecycle<InitProcess> for RuncInitLifecycle {
             })
             .collect())
     }
+
+    #[cfg(target_os = "linux")]
+    async fn pause(&self, p: &mut InitProcess) -> Result<()> {
+        let mut state = p.state.lock().await;
+        match *state {
+            Status::UNKNOWN => Err(other!("cannot pause an unknown process")),
+            Status::CREATED => Err(other!("cannot pause a created process")),
+            Status::RUNNING => {
+                *state = Status::PAUSING;
+                if let Err(e) = self.runtime.pause(p.id.as_str()).await {
+                    *state = Status::RUNNING;
+                    return Err(runtime_error(&self.bundle, e, "OCI runtime pause failed").await);
+                }
+                *state = Status::PAUSED;
+                Ok(())
+            }
+            Status::STOPPED => Err(other!("cannot pause a stopped process")),
+            Status::PAUSED => Err(other!("cannot pause a paused process")),
+            Status::PAUSING => Err(other!("cannot pause a pausing process")),
```

Comment:
All non-running cases can be covered with:

```rust
_ => Err(other!("cannot pause when in {:?} state", *state))
```

Reply:
Yes, that's right, thanks